streamwebs_frontend/streamwebs/templates/streamwebs/create_site.html

```diff
@@ -3,22 +3,136 @@
+  sites.push(temp);
+{% endfor %}
+
+$(document).ready(function() {
+  document.getElementById("site_form").reset();
+  var myLatLng = new google.maps.LatLng(43.8041, -120.5542);
+  var map = new google.maps.Map(document.getElementById('map'), {
+    zoom: 7,
+    center: myLatLng,
+    streetViewControl: false
+  });
+
+  google.maps.event.addListener(map, 'click', function(e){
+    placeMarker(e.latLng, map);
+  });
+
+  function clearMarkers(){
+    console.log(markers[0]);
+    if(markers[0]!== undefined){
+      console.log('yes');
+      markers[0].setMap(null);
+      markers.pop();
+    }
+  }
+
+  function placeMarker(position, map) {
+    clearMarkers();
+    var marker = new google.maps.Marker({
+      position: position,
+      map: map
+    });
+    markers.push(marker);
+    map.panTo(position);
+    $('#input_lat').val(position.lat());
+    $('#input_lng').val(position.lng());
+
+  }
+});
+function validateForm(){
+  var site_name = document.getElementById('id_site_name').value.toLowerCase();
+  var count = 0;
+  var found=sites.includes(site_name);
+  console.log(found);
+  // checks if the user has entered a name that is already in use
+  // regardless of the case
+  if(found){
+    Materialize.toast('That site name is already in use.', 3000, 'red');
+    count ++;
+  }
+  // checks to make sure user has selected a spot on the map
+  if (markers.length == 0){
+    Materialize.toast('Please select a location on the map.', 3000, 'red');
+    count ++;
+  }
+  // if either checks fail, form is not submitted and user is notified of
+  // the problems via toaster notification
+  if(count != 0){
+    return false;
+  }
+  else{
+    return true;
+  }
+}
+
+
+</script>
 {% endblock %}
 
-{% block title %}{% trans "Add a New Site" %}{% endblock %}
-{% block body_title %}{% trans "Add a New Site" %}{% endblock %}
+{% block body_title %} Add a New Site {% endblock %}
 
 {% block content %}
-  <form id='site_form' method='post' enctype="multipart/form-data">
-    {% csrf_token %}
-    {{ site_form.media }}
-    {{ site_form.as_p }}
-
-    <input class="btn waves-effect waves-light teal darken-4" type='submit'
-           name='submit' value='{% trans 'Add new site' %}'/>
-  </form>
+<form id='site_form' method='post' enctype="multipart/form-data"
+onsubmit="return validateForm()">
+{% csrf_token %}
+{{ site_form.media }}
+<div class="row">
+  <div class="input-field col s12">
+    <i class="material-icons prefix">flag</i>
+    <label for="{{ site_form.site_name.id_for_label }}">Site Name:</label>
+    {{ site_form.site_name }}
+    {{ site_form.site_name.errors | striptags }}
+
+  </div>
+</div>
+
+<div class="row">
+  <div class="input-field col s12">
+    <i class="material-icons prefix">sort</i>
+    <label for="{{ site_form.description.id_for_label }}">Description:</label>
+    {{ site_form.description }}
+    {{ site_form.description.errors | striptags }}
+  </div>
+</div>
+
+<input type="hidden" id="input_lat" name="lat" required>
```

Comment:
Please make these fields visible, it may be the case where a teach or student has the coordinates of a site, but doesn't know exactly where on the map to manually place a marker. The map should update if coordinates are entered manually, and the coordinate fields should update if a marker is placed or moved on the map.